src/main/kotlin/com/lykke/client/accounts/ClientAccountsCacheImpl.kt

```diff
@@ -2,16 +2,33 @@ package com.lykke.client.accounts
 
 import java.util.concurrent.ConcurrentHashMap
 
-class ClientAccountsCacheImpl(private val clientIdByWalletId: ConcurrentHashMap<String, String>): ClientAccountsCache {
+class ClientAccountsCacheImpl(private val clientIdByWalletId: ConcurrentHashMap<String, String>) : ClientAccountsCache {
+    private val walletsByClientId = ConcurrentHashMap<String, MutableSet<String>>()
+
+    init {
+        clientIdByWalletId.forEach { walletId, clientId ->
+            val wallets = walletsByClientId.getOrPut(clientId) { ConcurrentHashMap.newKeySet() }
+            wallets.add(walletId)
+        }
+    }
+
     internal fun delete(walletId: String) {
```

Comment:
Perhaps it is better to rename the method to `deleteWallet` now

Reply:
renamed